zebrad/src/components/sync/tests/vectors.rs

```diff
@@ -981,3 +987,61 @@ fn setup() -> (
         mock_chain_tip_sender,
     )
 }
+
+/// Test that the `end_of_support` function is working as expected.
+#[test]
+#[tracing_test::traced_test]
+#[should_panic(expected = "Zebra refuses to run if the release date is older than")]
+fn end_of_support_function() {
+    use std::str::FromStr;
+
+    let release_date: chrono::DateTime<chrono::Utc> =
+        chrono::DateTime::from_str(zebra_network::constants::RELEASE_DATE).unwrap();
+
+    // We are away from warn or panic
+    let no_warn = release_date
+        .checked_add_days(chrono::Days::new(
+            zebra_network::constants::EOS_PANIC_AFTER - 10,
+        ))
+        .unwrap();
+
+    sync::end_of_support(no_warn);
+    assert!(logs_contain("Checking if Zebra release is too old"));
+
+    // We are in warn range
+    let warn = release_date
+        .checked_add_days(chrono::Days::new(
+            zebra_network::constants::EOS_PANIC_AFTER - 3,
+        ))
+        .unwrap();
+
+    sync::end_of_support(warn);
+    assert!(logs_contain("Checking if Zebra release is too old"));
+    assert!(logs_contain(
+        "Your Zebra release is too old and it will stop running in"
+    ));
+
+    // We are in panic
+    let panic = release_date
+        .checked_add_days(chrono::Days::new(
+            zebra_network::constants::EOS_PANIC_AFTER + 1,
+        ))
+        .unwrap();
+
+    sync::end_of_support(panic);
+    assert!(logs_contain("Checking if Zebra release is too old"));
```

Comment:
Optional: it might help to split the checks that should panic from the checks that don't panic. That will make test failures easier to diagnose.